tests/multi_ptr/multi_ptr_common.h

```diff
@@ -163,6 +163,26 @@ struct check_pointer {
   }
 };
 
+/** @brief Initializes all elements in a container with 0
```

Comment:
This code is unclear to me. Where does the 0 come?
It is more efficient and clearer nowadays to use https://en.cppreference.com/w/cpp/utility/integer_sequence
Compilation time in O(1) instead of O(N).

Reply:
The 0's are expanded by recursively repeating 0 in the inheritance. For example, for `N == 2` we would have the inheritance chain

```
zero_init_container<T, 2>
  : zero_init_container<T, 1, 0>
  : zero_init_container<T, 0, 0, 0>
```
where `zero_init_container<T, 0, 0, 0>::value` ends up being `{T::value_type{0},T::value_type{0}}`.

We could use `std::integer_sequence` instead, but I don't see how we could construct with 0 easier using that. Granted, default initialization was the previous behavior so arguably any integer value would do.

Reply:
OK, why not.